MovieApp/PresentationLayer/HomeScreen/HomeScreenViewController.swift

```diff
@@ -111,69 +110,48 @@ class HomeScreenViewController: UIViewController {
                 })
             .store(in: &disposables)
 
-        topRatedMoviesCollectionView.onCategoryChanged = { [weak self] optionId in
-            self?.loadTopRatedMovies(for: optionId)
-        }
-
-        trendingMoviesCollectionView.onCategoryChanged = { [weak self] optionId in
-            self?.loadTrendingMovies(for: optionId)
-        }
-
-        popularMoviesCollectionView.onMovieFavorited = toggleFavorited
-        topRatedMoviesCollectionView.onMovieFavorited = toggleFavorited
-        trendingMoviesCollectionView.onMovieFavorited = toggleFavorited
-
-        popularMoviesCollectionView.onMovieSelected = selectedMovie
-        topRatedMoviesCollectionView.onMovieSelected = selectedMovie
-        trendingMoviesCollectionView.onMovieSelected = selectedMovie
-    }
-
-    private func loadTopRatedMovies(for optionId: Int, animated: Bool = true) {
-        presenter.getTopRatedMovies(for: optionId) { [weak self] result in
-            if case .success(let movies) = result {
-                self?.topRatedMoviesCollectionView.setData(movies, animated: animated)
-            } else {
-                self?.topRatedMoviesCollectionView.setData([], animated: animated)
+        topRatedMoviesCollectionView
+            .currentlySelectedCategory
+            .flatMap { [weak self] optionViewModel in
+                self?.presenter.topRatedMovies(for: optionViewModel.id) ?? .never()
             }
-        }
-    }
+            .sink(
+                receiveCompletion: { _ in },
+                receiveValue: { movieViewModels in
+                    self.topRatedMoviesCollectionView.setData(movieViewModels, animated: true)
+                })
+            .store(in: &disposables)
 
-    private func loadTrendingMovies(for optionId: Int, animated: Bool = true) {
-        guard let timeWindow = TimeWindowViewModel(rawValue: optionId) else { return }
+        trendingMoviesCollectionView
+            .currentlySelectedCategory
+            .flatMap { [weak self] optionViewModel -> AnyPublisher<[MovieViewModel], Error> in
+                guard let timeWindow = TimeWindowViewModel(rawValue: optionViewModel.id)
```

Comment:
`let timeWindow = TimeWindowViewModel(rawValue: optionViewModel.id)` in new line